src/main/java/net/javadiscord/javabot/systems/help/HelpListener.java

```diff
@@ -54,40 +48,58 @@ public class HelpListener extends ListenerAdapter implements ButtonHandler {
 	private final long SUGGEST_CLOSE_TIMEOUT = 5 * 60_000L;//5 minutes
 	private final Map<Long, Long> recentlyCloseSuggestedPosts = new LinkedHashMap<>(8, 0.75f, true) {
 		@Override
 		protected boolean removeEldestEntry(Map.Entry<Long, Long> eldest) {
 			return System.currentTimeMillis() > eldest.getValue() || size() >= 32;
 		}
 	};
-	private final UserPreferenceService preferenceService;
+	private final AutoCodeFormatter autoCodeFormatter;
+
+	/**
+	 * The constructor of this class, which is set corresponding to lombok.
+	 *
+	 * @param autoMod                   automod reference.
+	 * @param preferenceService         UserPreferenceService reference
+	 * @param botConfig                 BotConfig reference
+	 * @param helpTransactionRepository HelpAccountRepository reference
+	 * @param helpAccountRepository     HelpAccountRepository reference
+	 * @param dbActions                 DbActions reference
+	 * @param experienceService         ExperienceService reference
+	 */
+	public HelpListener(AutoMod autoMod, UserPreferenceService preferenceService, BotConfig botConfig, HelpAccountRepository helpAccountRepository, HelpTransactionRepository helpTransactionRepository, HelpExperienceService experienceService, DbActions dbActions) {
+		this.preferenceService = preferenceService;
+		this.botConfig = botConfig;
+		this.helpAccountRepository = helpAccountRepository;
+		this.helpTransactionRepository = helpTransactionRepository;
+		this.experienceService = experienceService;
+		this.dbActions = dbActions;
+		this.autoCodeFormatter = new AutoCodeFormatter(autoMod, botConfig, preferenceService);
+	}
 
 	@Override
 	public void onMessageReceived(@NotNull MessageReceivedEvent event) {
+		handleMessageEvent(event);
+		autoCodeFormatter.handleMessageEvent(event);
+	}
+
+	private void handleMessageEvent(@NotNull MessageReceivedEvent event) {
 		// listen for the posts initial message. Why this has to be done is further described in the JDA Discord:
 		// https://canary.discord.com/channels/125227483518861312/1053705384466059354/1053705384466059354
 		if (newThreadChannels.contains(event.getChannel().getIdLong())) {
 			HelpConfig config = botConfig.get(event.getGuild()).getHelpConfig();
 			ThreadChannel post = event.getChannel().asThreadChannel();
 			// send post buttons
-			post.sendMessageComponents(ActionRow.of(
-					createCloseSuggestionButton(post),
-					Button.secondary(ComponentIdBuilder.build(HelpManager.HELP_GUIDELINES_IDENTIFIER), "View Help Guidelines")
-			)).queue(success -> post.sendMessageFormat(config.getReservedChannelMessageTemplate(), UserSnowflake.fromId(post.getOwnerId()).getAsMention(), config.getInactivityTimeoutMinutes()).queue());
+			post.sendMessageComponents(ActionRow.of(createCloseSuggestionButton(post), Button.secondary(ComponentIdBuilder.build(HelpManager.HELP_GUIDELINES_IDENTIFIER), "View Help Guidelines"))).submit();
```

Comment:
Why are you using `submit()` instead of `queue()` here? You aren't using the `CompletableFuture`.

Reply:
using it to force blocking, since I had the problem that AutoCodeFormatter's messages would "arrive" before HelpListener's.

Reply:
`submit()` doesn't block. Also, JDA listeners shouldn't block.

Reply:
**will update